pygeoapi/provider/base.py

```diff
@@ -92,6 +92,17 @@ def get_fields(self):
 
         raise NotImplementedError()
 
+    @property
```

Comment:
The response should be a dict of dicts, so:

```python
    @property
    def fields(self) -> dict:
        """
        Store provider field information (names, types)

        Example response: {'field1': {'type': 'string'}}

        :returns: dict of dicts (field names and their associated JSON Schema definitions)
        """

        if hasattr(self, '_fields'):
            return self._fields
        else:
            return self.get_fields()
```

Given the above, a provider has the option to either set `self._fields` on init or to lazy load.  Then, from the downstream caller, the `.fields` accessor is opaque/consistent.